File: src/compiler/typescript_to_rust/generators/ic_object/functions/stable_grow.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionStableGrow(): Rust {
    return /* rust */ `
        fn _azle_ic_stable_grow(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context,
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            let new_pages: u32 = _aargs
                .get(0)
                .unwrap()
                .clone()
                .azle_try_from_js_value(_context)
                .unwrap();
            Ok(ic_cdk::api::stable::stable_grow(new_pages).azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate/src/body/ic_object/functions/stable_grow.rs
pub fn generate() -> proc_macro2::TokenStream {
    quote::quote! {
        fn stable_grow(
            _this: &boa_engine::JsValue,
            aargs: &[boa_engine::JsValue],
            context: &mut boa_engine::Context,
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            let new_pages: u32 = aargs
                .get(0)
                .unwrap()
                .clone()
                .try_from_vm_value(&mut *context)
                .unwrap();
            Ok(ic_cdk::api::stable::stable_grow(new_pages).try_into_vm_value(context).unwrap())
        }
    }
}
